torch/csrc/jit/codegen/onednn/interface.cpp

```diff
@@ -0,0 +1,145 @@
+#include <torch/csrc/jit/jit_log.h>
+#include <oneapi/dnnl/dnnl_graph.hpp>
+#include <torch/csrc/jit/codegen/onednn/defer_size_check.h>
+#include <torch/csrc/jit/codegen/onednn/graph_fuser.h>
+#include <torch/csrc/jit/codegen/onednn/guard_shape.h>
+#include <torch/csrc/jit/codegen/onednn/interface.h>
+#include <torch/csrc/jit/codegen/onednn/kernel.h>
+#include <torch/csrc/jit/codegen/onednn/layout_propagation.h>
+#include <torch/csrc/jit/codegen/onednn/prepare_binary.h>
+#include <torch/csrc/jit/passes/decompose_ops.h>
+#include <torch/csrc/jit/passes/pass_manager.h>
+#include <torch/csrc/jit/passes/remove_mutation.h>
+#include <torch/csrc/jit/passes/tensorexpr_fuser.h>
+#include <torch/csrc/jit/runtime/custom_operator.h>
+#include <torch/csrc/jit/runtime/graph_executor.h>
+#include <torch/csrc/jit/runtime/operator_options.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace onednn {
+
+void fuseGraph(std::shared_ptr<Graph>& g) {
+  // Follow the process of the tensorexpr_fuser in profiling mode:
+  // Remove prim::profile nodes and embed the profile info directly in the
+  // IR in value types to avoid breaking the fusion patterns.
+  // Will add shape guard after LLGA optimization passes and
+  // wipe the tensor type information from the IR, so that it's not
+  // accidentally used by any other pass.
+
+  // We rely on the shape specialization and shape guard to ensure the validity
+  // of the cached compilation in the kernel, thus only support profiling mode.
+  // TODO: add check on oneDNNFusionGroup to ensure allShapesAreKnown on nodes
+  // to fuse: torch/csrc/jit/passes/tensorexpr_fuser.cpp: allShapesAreKnown
+  if (getProfilingMode()) {
+    GRAPH_DUMP(
+        "Before RemoveProfileNodesAndSpecializeTypes. Beginning of LLGA "
+        "optimization pass",
+        g);
+    RemoveProfileNodesAndSpecializeTypes(g);
+    GRAPH_DUMP(
+        "After RemoveProfileNodesAndSpecializeTypes. Before mutation removal",
+        g);
+
+    RemoveTensorMutation(g);
+    RemoveListMutation(g);
+    GRAPH_DUMP("After mutation removal. Before DecomposeOps", g);
+    DecomposeOps(g);
```

Comment:
We probably don't want to do this here because it's going to remove type annotations from nodes. What decompositions do you need here?

Reply:
Thanks, I'll get back to you soon on this. I think we're calling it just because `torch::jit::runNondiffOptimization()` & `torch::jit::ProfilingGraphExecutorImpl::runProfilingInsensitiveOptimizations` were used as reference. 😄